src/TestFramework/TestFramework/Assertions/Assert.IComparable.cs

```diff
@@ -0,0 +1,604 @@
+    /// </param>
+    /// <param name="actual">
+    /// The second value to compare. This is the value produced by the code under test.
+    /// </param>
+    /// <exception cref="AssertFailedException">
+    /// Thrown if <paramref name="expected"/> is not greater than <paramref name="actual"/>.
+    /// </exception>
+    public static void IsGreaterThan<T>(T expected, T actual)
+        where T : struct, IComparable<T>
+        => IsGreaterThan(expected, actual, string.Empty, null);
+
+    /// <summary>
+    /// Tests whether the first value is greater than the second value and throws an exception
+    /// if it is not.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of values to compare.
+    /// </typeparam>
+    /// <param name="expected">
+    /// The first value to compare. This is the value the test expects to be greater.
+    /// </param>
+    /// <param name="actual">
+    /// The second value to compare. This is the value produced by the code under test.
+    /// </param>
+    /// <param name="message">
+    /// The message to include in the exception when <paramref name="expected"/>
+    /// is not greater than <paramref name="actual"/>. The message is shown in
+    /// test results.
+    /// </param>
+    /// <exception cref="AssertFailedException">
+    /// Thrown if <paramref name="expected"/> is not greater than <paramref name="actual"/>.
+    /// </exception>
+    public static void IsGreaterThan<T>(T expected, T actual, string? message)
+        where T : struct, IComparable<T>
+        => IsGreaterThan(expected, actual, message, null);
+
+    /// <summary>
+    /// Tests whether the first value is greater than the second value and throws an exception
+    /// if it is not.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of values to compare.
+    /// </typeparam>
+    /// <param name="expected">
+    /// The first value to compare. This is the value the test expects to be greater.
+    /// </param>
+    /// <param name="actual">
+    /// The second value to compare. This is the value produced by the code under test.
+    /// </param>
+    /// <param name="message">
+    /// The message to include in the exception when <paramref name="expected"/>
+    /// is not greater than <paramref name="actual"/>. The message is shown in
+    /// test results.
+    /// </param>
+    /// <param name="parameters">
+    /// An array of parameters to use when formatting <paramref name="message"/>.
+    /// </param>
+    /// <exception cref="AssertFailedException">
+    /// Thrown if <paramref name="expected"/> is not greater than <paramref name="actual"/>.
+    /// </exception>
+    public static void IsGreaterThan<T>(T expected, T actual, [StringSyntax(StringSyntaxAttribute.CompositeFormat)] string? message, params object?[]? parameters)
+        where T : struct, IComparable<T>
+    {
+        if (expected.CompareTo(actual) > 0)
```

Comment:
Please verify all compare calls in this file because it seems you have reversed the expected order. `Assert.IsGreaterThan(10, 5)` should fail but `Assert.IsGreaterThan(5, 10)` should succeed.

Basically `expected` is what to compare to and `actual` is the subject value